sources/packages/api/src/route-controllers/student/models/student.dto.ts

```diff
@@ -59,14 +60,13 @@ export interface SearchStudentRespDto {
 }
 
 /**
- * DTO Object for student restriction.
+ * DTO for student restriction.
  * This object is returned by controller.
+ * Here the key is the restrictionCode and value
+ * is the respective notificationType.
  */
-export interface StudentRestrictionDTO {
-  hasRestriction: boolean;
-  hasFederalRestriction: boolean;
-  hasProvincialRestriction: boolean;
-  restrictionMessage: string;
+export class StudentRestrictionAPIOutDTO {
+  [key: string]: RestrictionNotificationType;
```

Comment:
Why did we keep the key as dynamic here? I see the key is consisting of restriction code, what is benefit of having it dynamic.?

Reply:
It was a misunderstanding (that I believe that I caused), Ann will change it. I added the comment to the controller consuming it.